File: src/main/java/com/flowcommerce/core/support/error/ErrorMessage.java
package com.flowcommerce.core.support.error;

public record ErrorMessage<T>(String code, String message, T data) {

    public ErrorMessage(ErrorType errorType, T data) {
        this(errorType.getCode().name(), errorType.getMessage(), data);
    }

    public ErrorMessage(ErrorType errorType) {
        this(errorType, null);
    }
}

File: src/main/kotlin/com/flowcommerce/core/support/error/ErrorMessage.kt
package com.flowcommerce.core.support.error

data class ErrorMessage<T>(
    val code: String,
    val message: String,
    val data: T?
) {
    constructor(errorType: ErrorType, data: T?) : this(
        code = errorType.code.name,
        message = errorType.message,
        data = data
    )

    constructor(errorType: ErrorType) : this(errorType, null)
}
